components/eamxx/scripts/test_all_scream.py

```diff
@@ -386,6 +386,15 @@ def generate_cmake_config(self, test, for_ctest=False):
         if "SCREAM_DYNAMICS_DYCORE" not in custom_opts_keys:
             result += " -DSCREAM_DYNAMICS_DYCORE=HOMME"
 
+        # For the compute-sanitizer tool 'racecheck', if no option --racecheck-num-workers
+        # is provided, it will attempt to use all threads available on node. This can cause
+        # issues if other test cases are being run in parallel. If the option was not specified,
+        # limit the number of threads availible to racecheck to the number of compile resources.
+        if self._parallel and isinstance(test, CSR) and not '--racecheck-num-workers' in result:
+            new_option = ' --racecheck-num-workers=' + str(test.compile_res_count)
+            index = result.index('--tool=racecheck') + len('--tool=racecheck')
```

Comment:
I see this in test_factory.py:

```python
    ("EKAT_COMPUTE_SANITIZER_OPTIONS", "'--tool=racecheck --racecheck-detect-level=error'")],
```
So I am thinking that this call to `index` may fail... Why not sticking with the version without `'`, and just look for `--tool-racecheck`?

Reply:
I think python `str1.index(str2)` takes in a strings, no? That's why I added the `''`, so I'm looking for --tool=racecheck. But my python is rusty so I may misunderstand your question..

Reply:
Unless you mean the `'--tool=racecheck --racecheck-detect-level=error'`. If you don't include the outer quotes, cmake only picks up `EKAT_COMPUTE_SANITIZER_OPTIONS=--tool=racecheck` and you get an error
```
CMake Error: Unknown argument --racecheck-detect-level=error
```

Reply:
Ah, sorry, I mixed up the quotes looking at this and the mods in test_factory.py. Yes, this is fine.